engine-standalone-storage/src/relayer_db/types.rs

```diff
@@ -68,35 +69,81 @@ pub struct BlockRow {
     pub receipts_root: H256,
 }
 
-impl From<postgres::Row> for BlockRow {
-    fn from(row: postgres::Row) -> Self {
-        let chain: i32 = row.get("chain");
-        let id: i64 = row.get("id");
+struct BlockRowSize(i32);
+
+impl TryFrom<BlockRowSize> for u32 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowSize) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+struct BlockRowChain(i32);
+
+impl TryFrom<BlockRowChain> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowChain) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+struct BlockRowId(i64);
+
+impl TryFrom<BlockRowId> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowId) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+struct BlockRowBlock(i64);
+
+impl TryFrom<BlockRowBlock> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: BlockRowBlock) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
+    }
+}
+
+impl TryFrom<postgres::Row> for BlockRow {
+    type Error = postgres::Error;
+
+    fn try_from(row: postgres::Row) -> Result<Self, Self::Error> {
+        let chain: BlockRowChain = BlockRowChain(row.get("chain"));
+        let id: BlockRowId = BlockRowId(row.get("id"));
         let hash = get_hash(&row, "hash");
         let near_hash: Option<&[u8]> = row.get("near_hash");
         let timestamp = get_timestamp(&row, "timestamp");
-        let size: i32 = row.get("size");
+        let size: BlockRowSize = BlockRowSize(row.get("size"));
         let gas_limit = get_numeric(&row, "gas_limit");
         let gas_used = get_numeric(&row, "gas_used");
         let parent_hash = get_hash(&row, "parent_hash");
         let transactions_root = get_hash(&row, "transactions_root");
         let state_root = get_hash(&row, "state_root");
         let receipts_root = get_hash(&row, "receipts_root");
 
-        Self {
-            chain: chain as u64,
-            id: id as u64,
+        Ok(Self {
+            chain: u64::try_from(chain).unwrap(),
```

Comment:
You can change the type of Error for `TryFromIntError` and then use error propagation (`?`) here.

Reply:
After trials and errors, If one makes a new type that wraps `RowIter` type then add error handler for the crate::Error it would work. I need to see how much codes will be effected if I add the change.